app/ts/components/subcomponents/ConfigureRpcConnection.tsx

```diff
@@ -21,9 +24,68 @@ const ConfigureRpcContext = createContext<ConfigureRpcContext | undefined>(undef
 const RpcQueryProvider = ({ children }: { children: ComponentChildren }) => {
 	const { value: rpcQuery, waitFor, reset: resetRpcQuery } = useAsyncState<Network>()
 
+	const checkServerAvailability = async (url: string) => {
+		try {
+			const provider = new JsonRpcProvider(url)
+			return await provider.getNetwork()
+		} catch {
+			throw new Error('Server should be reachable')
+		}
+	}
+
+	const validateEthSimulateSupport = async (rpcEntry: RpcEntry) => {
+		// test eth_simulate request
+		const requestHandler = new EthereumJSONRpcRequestHandler(rpcEntry, true)
+		const ethSimulateV1ParamObject: EthSimulateV1Params['params'][0] = {
+			blockStateCalls: [{
+				blockOverrides: {
+					baseFeePerGas: 0x9n
+				},
+				stateOverrides: {
+					'0xc000000000000000000000000000000000000000': {
+						balance: 0x1312d0000n,
+					}
+				},
+				calls: [
+					{
+						from: 0xc000000000000000000000000000000000000000n,
+						to: 0xc000000000000000000000000000000000000000n,
+						value: 0x1n,
+						maxFeePerGas: 0xfn,
+					}
+				]
+			}],
+			validation: true,
+			traceTransfers: true
+		}
+
+		try {
+			const serializedResult = await requestHandler.jsonRpcRequest({
+				method: 'eth_simulateV1',
+				params: [ethSimulateV1ParamObject, 'latest']
+			})
+
+			const parsedResult = ethSimulateV1Result.parse(serializedResult)
+
+			for (const { calls } of parsedResult) {
```

Comment:
there should be just one block, with one call with one log. No need to loop over all of them

Reply:
switched to index reference and moved to a more readable function on [c996322](https://github.com/DarkFlorist/TheInterceptor/pull/1030/commits/c996322cae0016726ddc011ceaeea804a382148c)